src/main/kotlin/FilePathGenerator.kt

```diff
@@ -0,0 +1,103 @@
+import java.io.File
+import java.lang.IllegalArgumentException
+
+enum class SupportedExtensions(val ext: String) {
+    WAV("wav"),
+    MP3("mp3"),
+    JPEG("jpeg"),
+    JPG("jpg"),
+    TR("tr");
+
+    companion object {
+        private fun getSupportedContainers(): Array<SupportedExtensions> = arrayOf(TR)
+        private fun getCompressedTypes(): Array<SupportedExtensions> = arrayOf(MP3, JPEG, JPG)
+
+        fun isSupportedExtension(ext: String): Boolean = values().any { it.name == ext.toUpperCase() }
+
+        fun isSupportedContainer(ext: String): Boolean = getSupportedContainers().contains(valueOf(ext.toUpperCase()))
+        fun isCompressedType(ext: String): Boolean = getCompressedTypes().contains(valueOf(ext.toUpperCase()))
+    }
+}
+
+enum class Groupings(val grouping: String) {
+    BOOK("book"),
+    CHAPTER("chapter"),
+    VERSE("verse"),
+    CHUNK("chunk");
+
+    companion object {
+        fun isSupportedGrouping(grouping: String): Boolean = values().any { it.name == grouping.toUpperCase() }
+    }
+}
+
+enum class MediaQuality(val quality: String) {
+    HI("hi"),
+    LOW("low");
+
+    companion object {
+        fun isSupportedQuality(quality: String): Boolean = values().any { it.name == quality.toUpperCase() }
+    }
+}
+
+object FilePathGenerator {
+
+    fun createPathFromFile(
+        inputFile: File,
+        languageCode: String,
+        dublinCoreId: String,
+        grouping: String,
+        projectId: String = "",
+        mediaExtension: String = "",
+        mediaQuality: String = ""
+    ): String {
+
+        validateInput(inputFile, languageCode, dublinCoreId, grouping, mediaExtension, mediaQuality)
+
+        val projectPath = if (projectId.isBlank()) "" else "$projectId/"
+        val compressionQuality = if (mediaQuality.isBlank()) "hi" else mediaQuality
+
+        var path = "$languageCode/$dublinCoreId/${projectPath}CONTENTS/${inputFile.extension}/"
```

Comment:
I'd still prefer to see full string templates, not if/else statements and string concatenation